src/workerd/io/compatibility-date.capnp

```diff
@@ -517,4 +517,10 @@ struct CompatibilityFlags @0x8f8c1b68151b6cef {
   # Enables bypassing FL by translating pipeline tunnel configuration to subpipeline.
   # This flag is used only by the internal repo and not directly by workerd.
 
+  allowCustomPorts @55 :Bool
+    $compatEnableFlag("allow_custom_ports")
+    $compatEnableDate("2024-08-05");
```

Comment:
We should define this as experimental first, implement it and verify that it works as expected, and only then go back and set a default-on date. Otherwise there's a risk that it doesn't actually work yet on the default-on date and then we're in big trouble.

```suggestion
    $experimental;
```

Reply:
Makes sense, I amended it in.